components/threading/global_threads/global_threads.tsx

```diff
@@ -108,15 +108,18 @@ const GlobalThreads = () => {
         return selectedThreadId && list.length === 1 && list.includes(selectedThreadId);
     };
 
+    const shouldLoadThreads = (counts && counts.total !== 0) && (isEmpty(threadIds) || isOnlySelectedThreadInList(threadIds));
+    const shouldLoadUnreadThreads = (counts && counts.total_unread_threads !== 0) && (isEmpty(unreadThreadIds) || isOnlySelectedThreadInList(unreadThreadIds));
```

Comment:
Nit: Probably not much gain, but this is calculated on every render. If you move it to the effect body, it is calculated only when the effect is triggered. Depending on the cost of `isOnlySelectedThreadInList` may make more or less sense.

Reply:
I would think that `isOnlySelectedThreadInList` is not that costly.

It could be changed though to 
```js
         return selectedThreadId && list.length === 1 && list[0] === selectedThreadId; 
```